src/components/library-camera/screens/capture-photo.js

```diff
@@ -0,0 +1,138 @@
+import React, {useEffect, useRef, useState} from 'react';
+import {Alert, StyleSheet, TouchableOpacity, View} from 'react-native';
+import {RNCamera} from 'react-native-camera';
+import {PERMISSIONS, request, RESULTS} from 'react-native-permissions';
+
+import {container, fs20} from '../../../assets/styles/global.js';
+
+import navigationRoutes from '../../../constants/navigation-routes';
+import {
+  flashOff,
+  flashOn,
+  camera,
+  loop,
+} from '../../../constants/font-codes.js';
+
+import {GoBack, Navigate} from '../../../helpers/routes.js';
+import {IconIcomoon} from '../../shared/custom-icons.js';
+
+const {CREATE_BOOK} = navigationRoutes;
+
+const {
+  FlashMode: {on: flashModeOn, off: flashModeOff},
+  Type: {back: backCamera, front: frontCamera},
+} = RNCamera.Constants;
+
+const {isIOS, emptyArr} = global;
+
+const cameraOptions = {quality: 0.5, base64: true};
+
+const androidCameraPermissionOptions = {
+  title: 'Permission to use camera',
+  message: 'We need your permission to use your camera',
+  buttonPositive: 'Ok',
+  buttonNegative: 'Cancel',
+};
+
+const CapturePhoto = ({navigation}) => {
+  const [cameraAccess, setCameraAccess] = useState(false);
+  const [isFlashOn, setIsFlashOn] = useState(false);
+  const [isCameraTypeBack, setIsCameraTypeBack] = useState(true);
+
+  const cameraRef = useRef(null);
+
+  useEffect(() => {
+    request(isIOS ? PERMISSIONS.IOS.CAMERA : PERMISSIONS.ANDROID.CAMERA).then(
+      result => {
+        if (result == RESULTS.GRANTED) setCameraAccess(true);
+        else {
+          Alert.alert('Requires camera access');
+          GoBack(navigation);
+        }
+      },
+    );
+  }, emptyArr);
+
+  const takePicture = async () => {
+    if (cameraRef) {
+      const data = await cameraRef.current.takePictureAsync(cameraOptions);
+      Navigate(navigation, CREATE_BOOK, {imageUri: data.uri});
+    }
+  };
+
+  const toggleCamera = () => {
+    setIsCameraTypeBack(prevState => !prevState);
+  };
+
+  const toggleFlash = () => {
+    setIsFlashOn(prevState => !prevState);
+  };
+
+  const onBarCodeRead = e => {
+    Alert.alert('Barcode value is' + e.data, 'Barcode type is' + e.type);
+  };
+
+  const onGoogleVisionBarcodesDetected = ({barcodes}) => {
+    barcodes.forEach(barcode => console.warn(barcode.data));
+  };
+
+  const flashContent = isFlashOn ? flashOn : flashOff;
+  const flashMode = isFlashOn ? flashModeOn : flashModeOff;
+  const cameraType = isCameraTypeBack ? backCamera : frontCamera;
+
+  const {preview, btnContainer, capture} = styles;
+
+  return (
+    <View style={container}>
+      {cameraAccess && (
+        <>
+          <RNCamera
+            ref={cameraRef}
+            style={preview}
+            type={cameraType}
+            flashMode={flashMode}
+            onBarCodeRead={onBarCodeRead}
+            androidCameraPermissionOptions={androidCameraPermissionOptions}
+            captureAudio={false}
```

Comment:
Lets also add a view here that if there are no camera permissions. Then it should show "Permission denied" And an info -> "Please change the permissions in settings" and a button which would open the app settings